src/Lib/Element.php

```diff
@@ -109,18 +109,43 @@ public function hasOptions()
     }
 
     /**
+     * Returns whether this element conatins elements
+     * @return bool
+     */
+    public function hasElements()
+    {
+        return count($this->elements) > 0;
```

Comment:
:+1: On the switch. Because some elements can contain external elements / options, this isn't calculated until serialization so option 2 isn't possible.